apisix/plugins/dynamic-upstream.lua

```diff
@@ -282,61 +278,69 @@ local function set_upstream(upstream_info, ctx)
 end
 
 
-local function new_rr_obj(up_info)
+local function new_rr_obj(upstreams)
     local server_list = {}
-    for i = 1, #up_info, 2 do
-        server_list[tostring(i+1)] = up_info[i]
-        core.log.info("tostring(i+1): ", i+1, "up_info-weight: ", up_info[i])
+    for i, upstream_obj in ipairs(upstreams) do
+        if not upstream_obj.upstream then
+            upstream_obj.upstream = "empty_upstream"
```

Comment:
that is totally wrong, you update the original `conf`.
we should never to write the original `conf`.